### DISCUSSION TITLE
Does CKAN support data access upon request?

### DISCUSSION THREAD

**Discussion by Duragtan** [ckan/ckan#7659] [Q&A]
Hello all,

I am wondering if CKAN might be the right tool for my purposes. Can someone tell me if the following scenario is supported by CKAN:

> Person A has authored a data set, which he/she cannot easily disclose. Person A uploads the data set to CKAN, but as a private data set that others can only see the meta-data about. 
> Now Person B is interested in working with the data. He/she contacts Person A via a web-form in CKAN, requesting data access. Person B can now decide whether or not Person B should get access to the data set. If approved, Person B can access the full data on CKAN for a defined period. CKAN logs the entire interaction between the two Persons.
> Occasionally, Person A might only want to put an entry into CKAN describing the data set, but without actually uploading the data to CKAN (a pure meta-data entry). Would the process described above still work for a pure meta-data entry. (In that case, the two would, of course, have to find some other means for actually transferring the data.)

I have already found out about the ckanext-restricted module (https://github.com/EnviDat/ckanext-restricted), which seems to support at least part of what I want. Does anyone have any experiences with it?

Best

Soraltan

**Comment by rmosavi** [comment]
Hi Sorlatan, did you get answer for your scenario for the scenario that you set. I am thinking the same scenario for our group. If no, have found any other open source that can manage the scenario?

Best and looking forward to hearing from you
